dev-playground/public/samples/vectors

```diff
@@ -0,0 +1,144 @@
+content: [
+	{
+		text: [
+			'This ',
+			{text: 'is', color: 'green'},
+			' the first ',
+			{text: 'paragraph', color: 'red'}
+		]
+	},
+	{
+		canvas: [
+			{
+				type: 'rect',
+				x: 0,
+				y: 0,
+				w: 310,
+				h: 260,
+				r: 5,
+				dash: {length: 5},
+				// lineWidth: 10,
+				lineColor: 'blue',
+			},
+			{
+				type: 'rect',
+				x: 1,
+				y: 1,
+				w: 308,
+				h: 258,
+				r: 4,
+				lineColor: 'red',
+				color: '#ffffe0',
+			},
+			{
+				type: 'polyline',
+				lineWidth: 3,
+				closePath: true,
+				points: [{x: 10, y: 10}, {x: 35, y: 40}, {x: 100, y: 40}, {x: 125, y: 10}]
+			},
+			{
+				type: 'polyline',
+				lineWidth: 2,
+				color: 'blue',
+				lineColor: 'red',
+				points: [{x: 10, y: 110}, {x: 35, y: 140}, {x: 100, y: 140}, {x: 125, y: 110}, {x: 10, y: 110}]
+			},
+			{
+				type: 'line',
+				x1: 40, y1: 60,
+				x2: 260, y2: 60,
+				lineWidth: 3
+			},
+			{
+				type: 'line',
+				x1: 40, y1: 80,
+				x2: 260, y2: 80,
+				lineWidth: 10,
+				lineCap: 'round'
+			},
+			{
+				type: 'line',
+				x1: 40, y1: 100,
+				x2: 260, y2: 100,
+				lineWidth: 10,
+				lineCap: 'square'
+			},
+			{
+				type: 'ellipse',
+				x: 150, y: 140,
+				color: 'red',
+				fillOpacity: 0.5,
+				r1: 80, r2: 60
+			},
+			{
+				type: 'rect',
+				x: 150,
+				y: 200,
+				w: 150,
+				h: 50,
+			},
+			{
+				type: 'rect',
+				x: 10, y: 200, w: 100, h: 10,
+				linearGradient: ['red', 'blue']
+			},
+			{
+				type: 'rect',
+				x: 10, y: 215, w: 100, h: 10,
+				linearGradient: ['red', 'green', 'blue']
+			},
+			{
+				type: 'rect',
+				x: 10, y: 230, w: 100, h: 10,
+				linearGradient: ['red', 'yellow', 'green', 'blue']
+			},
+			//svg example:
+			//first: transformations
+			{
+				function: 'translate',
+				x: 250,
+				y: 400
+			},
+			{
+				function: 'scale',
+				x: 0.5,
+				y: 0.5
+			},
+			{
+				function: 'rotate',
+				angle: 45
+			},
+			//draw the svg
+			{
+				type: 'path',
+				lineWidth: 1,
+				color: 'blue',
+				lineColor: 'yellow',
+				d: 'M150 0 L75 200 L225 200 Z',
+			},
```

Comment:
Hi ! I was wondering if there was svg support for pdfMake then I saw you pull request. Thanks for doing it !

Now that I'm here, I find the API not declarative enough comparing to the pdfMake API, and too far from the svg one.
A better alternative would be to contain those transformations on the element being transformed instead of separate instructions :

As requested in #807
```js
{ type: 'rect', x: 150, y: 200, w: 150, h: 50, rotate:45, translate: {x: 10, y: 1}, scale: 1 }
```

However order of transforms do matters so maybe something like that would do the job:
```js
{ 
  type: 'rect', 
  x: 150,
  y: 200,
  w: 150,
  h: 50,
  transform(s): [
    {translate: {x: 10, y: 1}},
    {rotate: 45},
    {scale: 1},
  ],
}
```

Which is close to the svg API `transform="translate(10, 1) scale(-1, 1)"`

And if it is needed to apply the same operations to multiple elements maybe introduce a node of type `'group'` like the `<g>` node in svg ?

Last point, I we want a better svg support maybe introduce 2 other transforms : `skew`  (and `skewX` and `skewY`) and `matrix`. Skew is not implemented by pdfkit but there is a generic `transform` method that can take any matrix, and the matrices for skew are available here : https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/transform